pkg/cloudprovider/providers/azure/azure_instances.go

```diff
@@ -121,7 +121,20 @@ func (az *Cloud) InstanceExistsByProviderID(ctx context.Context, providerID stri
 
 // InstanceShutdownByProviderID returns true if the instance is in safe state to detach volumes
 func (az *Cloud) InstanceShutdownByProviderID(ctx context.Context, providerID string) (bool, error) {
-	return false, cloudprovider.NotImplemented
+
+	nodeName, err := az.vmSet.GetNodeNameByProviderID(providerID)
+
+	if err != nil {
+		return false, err
+	}
+
+	provisioningState, err := az.vmSet.GetProvisioningStateByNodeName(string(nodeName))
+
+	if err != nil {
+		return false, err
+	}
+
+	return provisioningState == "Stopped" || provisioningState == "Deallocated", nil
```

Comment:
I would suggest all using lower case to do string comparison, to prevent case change in the future.